app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt

```diff
@@ -130,13 +125,12 @@ class MainFragment : Fragment() {
                 if(grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED){
                     takePhoto()
                 } else {
-                    Toast.makeText(context, "Unable to take photo without permission", Toast.LENGTH_LONG).show()
+                    Toast.makeText(context, "Unable to take photo without permission, goto Apps & notifications within android settings to enable", Toast.LENGTH_LONG).show()
```

Comment:
My own suggestion just having this question brought up a lot when it comes to app permissions
After a user denies a app permission, they must re enable it themselves within their phone settings. Not many users know this, so I just added some info to the end of the Toast when user declines camera permissions

Reply:
I'd exercise caution on directing a user how to use the phone operating system, as that may change over time.

While you're here, though, I'd consider something that will align with best practices: externalize the string to strings.xml.